src/many-error/src/error.rs

```diff
@@ -45,6 +45,8 @@ macro_rules! many_error {
                         $v => Self::$name,
                     )*
                     x if x >= 0 => Self::ApplicationSpecific(x as u32),
+                    // TODO: need to clarify whether `x <= -10000` or `x <= 0 && x % 10000 == 0`
```

Comment:
@hansl Yuri was wondering this because the following comment in the code isn't clear enough. Can you shed some light, please? I *think* the following is wrong.

```
    // Negative 10000+ are reserved for attribute specified codes and are defined separately.
    // The method to use these is ATTRIBUTE_ID * -10000.
```

Reply:
Yeah, we should also update the spec (the file that should describe errors is missing from the spec, btw)

Reply:
The spec specifies the error format for now; that's a deliberate omission. We need to address this at some point, but errors are very general and we'll miss some for sure.

What is defined here is that there is a range of errors for an attribute of `(ATTRIBUTE_ID * -10000 .. (ATTRIBUTE_ID+1) * -10000(`.

Reply:
So attribute `55` would have the range of error codes `-55000 .. -55999` (inclusive).

Reply:
Ah, ok, then this PR should be correct.